openlibrary/plugins/importapi/code.py

```diff
@@ -338,24 +339,33 @@ def get_ia_record(metadata: dict) -> dict:
         lccn = metadata.get('lccn')
         subject = metadata.get('subject')
         oclc = metadata.get('oclc-id')
+        number_of_pages = metadata.get('imagecount')
         d = {
-            'title': metadata.get('title', ''),
+            'title': metadata.get('title', '').strip(),
             'authors': authors,
-            'publish_date': metadata.get('date'),
-            'publisher': metadata.get('publisher'),
+            'publish_date': metadata.get('date', '').strip(),
+            'publisher': metadata.get('publisher', '').strip(),
         }
         if description:
-            d['description'] = description
+            d['description'] = description.strip()
         if isbn:
             d['isbn'] = isbn
-        if language and len(language) == 3:
-            d['languages'] = [language]
+        if language:
+            if len(language) == 3:
+                d['languages'] = [language]
+
+            # Try converting the name of a language to its three character code.
+            # E.g. English -> eng.
+            elif lang_code := get_abbrev_from_full_lang_name(language):
+                d['languages'] = [lang_code]
```

Comment:
Nice! Could we log a warning if no matching language, or multiple matching languages are found?

Reply:
I went ahead and added this.